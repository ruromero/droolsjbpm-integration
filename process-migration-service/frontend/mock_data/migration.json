{
  "id": 99,
  "definition": {
    "execution": {
      "type": "ASYNC",
      "callbackUrl": "This is mockup response",
      "scheduledStartTime": null
    },
    "planId": 2,
    "processInstanceIds": [
      1
    ],
    "kieServerId": "sample-server"
  },
  "status": "CREATED",
  "createdAt": "2019-07-17T20:59:22.128Z"
}