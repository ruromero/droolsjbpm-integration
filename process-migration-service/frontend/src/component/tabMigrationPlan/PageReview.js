import React from "react";
import { Button } from "patternfly-react";
import PropTypes from "prop-types";

export default class PageReview extends React.Component {
  constructor(props) {
    super(props);
  }

  doExport = () => {
    const toExport = Object.assign({}, this.props.object);
    delete toExport.id;
    const atag = document.createElement("a");
    const exportableJSON = JSON.stringify(toExport, null, 2);
    const file = new Blob([exportableJSON], { type: "application/json" });
    atag.href = URL.createObjectURL(file);
    atag.download = this.props.exportedFileName + ".json";
    atag.click();
  };

  render() {
    return (
      <React.Fragment>
        <Button onClick={this.doExport}>Export</Button>
        <pre>{JSON.stringify(this.props.object, null, 2)}</pre>
      </React.Fragment>
    );
  }
}

PageReview.defaultProps = {
  exportedFileName: "exported"
};

PageReview.propTypes = {
  object: PropTypes.object.isRequired,
  exportedFileName: PropTypes.string
};
