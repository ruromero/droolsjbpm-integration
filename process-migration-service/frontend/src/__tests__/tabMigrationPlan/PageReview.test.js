import renderer from "react-test-renderer";
import React from "react";
import PageReview from "../../component/tabMigrationPlan/PageReview";
import migration from "../../../mock_data/migration.json";

test("PageReview renders correctly using snapshot", () => {
  const tree = renderer
    .create(<PageReview object={migration} exportedFileName="miration" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
