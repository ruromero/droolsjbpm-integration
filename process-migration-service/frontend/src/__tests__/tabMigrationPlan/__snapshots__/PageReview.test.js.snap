// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageReview renders correctly using snapshot 1`] = `
Array [
  <button
    className="btn btn-default"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Export
  </button>,
  <pre>
    {
  "id": 99,
  "definition": {
    "execution": {
      "type": "ASYNC",
      "callbackUrl": "This is mockup response",
      "scheduledStartTime": null
    },
    "planId": 2,
    "processInstanceIds": [
      1
    ],
    "kieServerId": "sample-server"
  },
  "status": "CREATED",
  "createdAt": "2019-07-17T20:59:22.128Z"
}
  </pre>,
]
`;
